### PR TITLE
Remove duplicating rule for cordova/www/css/PROJECT_NAME.css

### PR DESCRIPTION
**Pull request by dannywillems**
See explanation in `Makefile.mobile` (the commit).

### CODE REVIEW

File: template.distillery/Makefile.mobile [hunk 1]
```diff
@@ -216,11 +216,17 @@ CORDOVA_STATIC_FILES := \
...
+# Cordova www directory.
+# The file $(LOCAL_CSS), which is $(LOCAL_STATIC_CSS)/$(PROJECT_NAME).css, is
+# excluded, because it's producing another rule for $(APPCSS). By excluding this
+# file, the CSS file which will be retrieved from the server (remote or local
+# depending on the APP_REMOTE value) instead of copied from $(LOCAL_CSS).
 LOCAL_STATIC_FILES = $(patsubst \
     $(LOCAL_STATIC)/%, \
     $(CORDOVAPATH)/www/%, \
-    $(shell find $(LOCAL_STATIC) -type f))
+    $(shell find $(LOCAL_STATIC) -type f ! -name '$(PROJECT_NAME)' \
```
Comment (dannywillems) [inline template.distillery/Makefile.mobile]: @vasilisp It worked without `.css`. I changed it.

**Comment by dannywillems** [comment]
@vasilisp Sometimes, I have `$(LOCAL_STATIC_FILES)` so I need to investigate.

**Comment by dannywillems** [comment]
Can be merged.

File: template.distillery/Makefile.mobile [hunk 2]
```diff
@@ -216,11 +216,17 @@ CORDOVA_STATIC_FILES := \
...
+# The file $(LOCAL_CSS), which is $(LOCAL_STATIC_CSS)/$(PROJECT_NAME).css, is
+# excluded, because it's producing another rule for $(APPCSS). By excluding this
+# file, the CSS file which will be retrieved from the server (remote or local
+# depending on the APP_REMOTE value) instead of copied from $(LOCAL_CSS).
 LOCAL_STATIC_FILES = $(patsubst \
     $(LOCAL_STATIC)/%, \
     $(CORDOVAPATH)/www/%, \
-    $(shell find $(LOCAL_STATIC) -type f))
+    $(shell find $(LOCAL_STATIC) -type f ! -name '$(PROJECT_NAME).css' \
+                 ! -path '$(LOCAL_STATIC_CSS)'))
```
Comment (vasilisp) [inline template.distillery/Makefile.mobile]: Do you want to exclude everything under `$(LOCAL_STATIC_CSS)`?

Comment (dannywillems) [inline template.distillery/Makefile.mobile]: No, it only excludes the file `$(PROJECT_NAME).css`. Other files like `static/www/css/os.css` are copied.

Comment (dannywillems) [inline template.distillery/Makefile.mobile]: >  It excludes nothing but the directory $(LOCAL_STATIC_CSS).

From `man find`
```Man
-path pattern
  True if the pathname being examined matches pattern.
```
It's not excluding the directory. If the examinated file has `$(LOCAL_STATIC_CSS)` as pathname (`static/css`) and has `$(PROJECT_NAME).css` as basename, it excludes this file.

What I don't understand is why I need to add `!` before `$(LOCAL_STATIC_CSS)`. I thought
```
find [path] ! -name [basename] -path [pathname]
```
was equivalent to
```
find [path] (! -name [basename]) -and (-path [pathname])
```
but it seems the negation is propagate to all other conditions (hence I use another `!`).
I suppose the condition evaluation is right-to-left. So I suppose it's equivalent to
```
find [path] ! (-name [basename] -and -path [pathname])
```
Must be tested...

**Comment by dannywillems** [comment]
I close it. I don't think it's very useful.

**Pull request closed without merging**